assets/src/components/query/limit/index.js

```diff
@@ -31,5 +30,7 @@ export const QueryLimit = ( {
 			max={ max }
 			onChange={ onLimitChange }
 		/>
+	) : (
+		null
```

Comment:
I like the changes you made here to verify that onLimitChange is a function before passing along to the component, but I'm wondering whether we _need_ to do this.  When we do this type checking, there is no error in the console and thus no helpful indication for _why_ something isn't working.  So I think it may be actually more useful here if we do _no_ type checking for onChange here and simply pass it along.  If client code does not define a function for onChange then there will be an error thrown in the console with a stack trace that helps provide information about what caused the error.

I would consider this similar to how we throw uncaught exceptions in php except with javascript you don't have to throw the error because it automatically happens for syntax errors.

Now, if this was a prop passed through because of dynamic user facing controls I would agree its type needs to be checked.  But this is a prop passed through from actual code written so its the kind of error that would immediately surface if utilized incorrectly.

Incidentally this is why I think:

- WordPress GB team doesn't do extensive typechecking in their components (notice that `RangeControl` does not have type checking for `onChange`.
- Many js teams use typescript or flow because it provides typechecking outside of runtime while actually writing code.  I don't think we need to go there yet because its a whole other syntax to learn and will slow our progress even more.  

I'm not saying we drop type checking all-together.  I just think that for certain props passed along that are defined by code, if a console error produced by javascript would be sufficient to trace improper usage of a component then I think that should be sufficient.

Reply:
that all sounds logical... will change.